<launch>

    <arg name="namespace" default="sam"/>

    <group ns="$(arg namespace)">
    	<group ns="ctrl">
			<!-- Emergency surface action -->
			<!-- <node name="emergency_surface_action" pkg="sam_action_servers" type="emergency_surface_action.py" output="screen">
				<param name="emergency_topic" value="/$(arg namespace)/core/abort" />
				<param name="vbs_cmd_topic" value="/$(arg namespace)/core/vbs_cmd" />
				<param name="rpm1_cmd_topic" value="/$(arg namespace)/core/thruster1_cmd" />
				<param name="rpm2_cmd_topic" value="/$(arg namespace)/core/thruster2_cmd" />
				<param name="vbs_pid_enable_top" value="/$(arg namespace)/ctrl/vbs/pid_enable" />
			</node> -->

		<node name="goto_waypoint" pkg="floatsam_action_servers" type="usv_wp_goto_action.py" output="screen">
			<param name="base_frame" value="$$(arg namespace)/base_link"/>
			<param name="base_frame_2d" value="$(arg namespace)/base_link_2d"/>
			<param name="wp_tolerance" value="2."/>
			<param name="node_freq" value="40." />
			<param name="heading_setpoint_topic" value="/$(arg namespace)/ctrl/heading/setpoint" />
			<param name="heading_enable_topic" value="/$(arg namespace)/ctrl/heading/enable" />
			<param name="rpm_enable_top" value="/$(arg namespace)/ctrl/rpm/enable" />
			<param name="rpm_sp_top" value="/$(arg namespace)/ctrl/rpm/setpoint" />
		</node>
	</group>

	</group>


	<!-- Planned surface node -->
	<!-- <include file="$(find sam_action_servers)/launch/mission_complete_node.launch" >
		<arg name="robot_name" value="$(arg namespace)"/>	
	</include> -->

    
</launch>
